src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,48 +1,93 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo,
+  clickModal: () => void
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, clickModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasLoadingError, setHasIsLoadingError] = useState(false);
+
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      try {
+        const loadedUser = await getUser(userId);
+
+        setUser(loadedUser);
+        setIsLoading(true);
+        setHasIsLoadingError(false);
+      } catch (error) {
+        setIsLoading(true);
```

Comment:
why you setting isLoading to `true` on both `try` and `catch` blocks? what's the purpose? maybee it's enough just to set to `false` on `finally`?